app/services/teacher_training_public_api/sync_sites.rb

```diff
@@ -0,0 +1,106 @@
+module TeacherTrainingPublicAPI
+  class SyncSites
+    attr_reader :provider
+    attr_reader :course
+
+    include Sidekiq::Worker
+    sidekiq_options retry: 3, queue: :low_priority
+
+    def perform(provider_id, recruitment_cycle_year, course_id)
+      @provider = ::Provider.find(provider_id)
+      @course = ::Course.find(course_id)
+
+      sites = TeacherTrainingPublicAPI::Location.where(
+        year: recruitment_cycle_year,
+        provider_code: @provider.code,
+        course_code: @course.code,
+      ).includes(:location_status).paginate(per_page: 500)
```

Comment:
We probably don't need to paginate here, we're not going to get over 500 sites any time soon